src/lidar/read_rp_lidar.cpp

```diff
@@ -0,0 +1,82 @@
+#include "read_rp_lidar.h"
+
+#include "../navtypes.h"
+#include "../world_interface/world_interface.h"
+#include "../Constants.h"
+#include "PointCloudProcessing.h"
+#include "PointGenerator.h"
+
+#include <atomic>
+#include <mutex>
+#include <thread>
+#include <iostream>
+
+using namespace navtypes;
+
+bool lidar_initialized = false;
+RPLidar rp_lidar(Constants::Lidar::RP_PATH, Constants::Lidar::RPLIDAR_A1_BAUDRATE);
```

Comment:
We are probably going to use the S1s over the A1s on the real rover - we got a mount made for them already and everything.
```suggestion
RPLidar rp_lidar(Constants::Lidar::RP_PATH, Constants::Lidar::RPLIDAR_S1_BAUDRATE);
```